Foundation/ISO8601DateFormatter.swift

```diff
@@ -79,8 +79,44 @@ open class ISO8601DateFormatter : Formatter, NSSecureCoding {
         super.init()
     }
     
-    public required init?(coder aDecoder: NSCoder) { NSUnimplemented() }
-    open override func encode(with aCoder: NSCoder) { NSUnimplemented() }
+    public required init?(coder aDecoder: NSCoder) {
+        guard aDecoder.allowsKeyedCoding else {
+            fatalError("ISO8601DateFormatter cannot be decoded by non-keyed archivers")
+        }
+        
+        let options = Options(rawValue: UInt(aDecoder.decodeInteger(forKey: "NS.formatOptions")))
+        let timeZone: NSTimeZone?
+        
+        if aDecoder.containsValue(forKey: "NS.timeZone") {
+            if let tz = aDecoder.decodeObject(of: NSTimeZone.self, forKey: "NS.timeZone") {
+                timeZone = tz
+            } else {
+                aDecoder.failWithError(CocoaError(.coderReadCorrupt, userInfo: [ NSLocalizedDescriptionKey: "Time zone was corrupt while decoding ISO8601DateFormatter" ]))
+                return nil
+            }
+        } else {
+            timeZone = nil
+        }
+        
+        self.formatOptions = options
```

Comment:
I would combine this with the `options` declaration, i.e.

`self.formatOptions = Options(rawValue: UInt(aDecoder.decodeInteger(forKey: "NS.formatOptions")))`

Or at least put this line right after the `let options =` one.

Reply:
Makes sense.